File: tests/Generator/resource/visualbasic/complex/SecurityOAuth.vb
Imports System.Text.Json.Serialization

Public Class SecurityOAuth
    Inherits Security
    <JsonPropertyName("authorizationUrl")>
    Public Property AuthorizationUrl As Nullable(String)

    <JsonPropertyName("scopes")>
    Public Property Scopes As Nullable(String())

    <JsonPropertyName("tokenUrl")>
    Public Property TokenUrl As Nullable(String)

End Class

File: tests/Generator/resource/rust/complex/security_o_auth.rs
use serde::{Serialize, Deserialize};
use security::Security;

#[derive(Serialize, Deserialize)]
pub struct SecurityOAuth {
    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "authorizationUrl")]
    authorization_url: Option<String>,

    #[serde(rename = "scopes")]
    scopes: Option<Vec<String>>,

    #[serde(rename = "tokenUrl")]
    token_url: Option<String>,

}

